vizro-core/src/vizro/models/_dashboard.py

```diff
@@ -109,26 +129,62 @@ def _make_page_layout(self, page: Page):
         # Shared across pages but slightly differ in content. These could possibly be done by a clientside
         # callback instead.
         page_title = html.H2(children=page.title, id="page_title")
-        navigation: _NavBuildType = cast(Navigation, self.navigation).build(active_page_id=page.id)
-        nav_bar = navigation["nav_bar_outer"]
-        nav_panel = navigation["nav_panel_outer"]
+        nav_content: _NavBuildType = cast(Navigation, self.navigation).build(active_page_id=page.id)
+        nav_bar = nav_content["nav_bar_outer"]
+        nav_panel = nav_content["nav_panel_outer"]
 
         # Different across pages
         page_content: _PageBuildType = page.build()
         control_panel = page_content["control_panel_outer"]
-        component_container = page_content["component_container_outer"]
-
+        components = page_content["component_container_outer"]
+        return {
+            "logo": logo,
+            "dashboard_title": dashboard_title,
+            "theme_switch": theme_switch,
+            "page_title": page_title,
+            "nav_bar": nav_bar,
+            "nav_panel": nav_panel,
+            "control_panel": control_panel,
+            "components": components,
+        }
+
+    def _arrange_left_side(self, page_divs: PageDivs):
         # Arrangement
-        header = html.Div(children=[page_title, theme_switch], className="header", id="header_outer")
-        nav_control_elements = [dashboard_title, nav_panel, control_panel]
-        nav_control_panel = (
-            html.Div(nav_control_elements, className="nav_control_panel")
-            if any(not getattr(element, "hidden", False) for element in nav_control_elements)
+        left_header_divs = [page_divs["dashboard_title"]]
+        left_sidebar_divs = [page_divs["nav_bar"]]
+
+        if getattr(page_divs["nav_bar"], "hidden", False) is False:
+            left_sidebar_divs.insert(0, page_divs["logo"])
+        else:
+            left_header_divs.insert(0, page_divs["logo"])
```

Comment:
Not really pretty, so I am up for improvement suggestions if there are some :) But in general I realised that we could enable all the common 13 layouts as long as we can move the logo to different containers and considering two alternative design proposals I have

Reply:
In general I think we will find this easier if instead of manipulating lists/dictionaries we are manipulating the actual Dash component tree.

If this `getattr(element, "hidden", False)` thing becomes used a lot then let's put it in a helper function also.

Reply:
Currently implemented as `_get_hideable_parent_div` function

Reply:
I went with your proposal from today and implemented a `_is_hidden` check [here](https://github.com/mckinsey/vizro/pull/205/commits/9210ca302411694fdad867267b1abdf0d4bcfac1) - there were definitely pros/cons in this new check compared to the previous function. 

- A pro was definitely that it simplified the code and the tests
- A minor con was that the children were indeed visible in the DOM but also as a hidden attribute (didn't bother me that much)
- A major con was that the CSS stylings with `XXXX:not(:empty)` didn't work properly anymore as hidden children do not count as empty (therefore the containers didn't collapse properly and the dividers were visible even though there was no content)

There is an important CSS fix I've added which I'll explain later on, but this basically got rid of the major con here + fixed some other ambiguous behaviour which we had because we sometimes return `None` or `html.Div(hidden=True)`. So I sticked with this approach 👍

Reply:
This seems like a great solution, thank you for working through it 👍

Reply:
The whole thing looks much nicer than the original bit of code we were commenting on here with the `getattr` and the `insert` 💯